gnd/src/main/java/com/google/android/gnd/ui/MapIcon.java

```diff
@@ -17,80 +17,54 @@
 import com.google.android.gms.maps.model.BitmapDescriptor;
 import com.google.android.gms.maps.model.BitmapDescriptorFactory;
 import com.google.android.gnd.R;
-import com.google.android.gnd.ui.util.ViewUtil;
 
 // TODO: Move to common, rename Marker.
 public class MapIcon {
-  private static final String TAG = MapIcon.class.getSimpleName();
   private final Context context;
-  private BitmapDrawable drawable;
-  private int color;
+  private final int color;
 
-  public MapIcon(Context context, @Nullable String iconId, @Nullable String color) {
+  public MapIcon(Context context, @Nullable String colorHexCode) {
     this.context = context;
-    this.color = getIconColor(context, color);
-    int resourceId = getResourceId(context, iconId);
-    this.drawable = (BitmapDrawable) ContextCompat.getDrawable(context, resourceId);
+    this.color = getIconColor(context, colorHexCode);
   }
 
-  private static int getIconColor(Context context, String color) {
-    try {
-      return Color.parseColor(color);
-    } catch (Exception e) {
-      return context.getResources().getColor(R.color.colorMapAccent);
-    }
+  public BitmapDescriptor getBitmap() {
+    Drawable outline = AppCompatResources.getDrawable(context, R.drawable.ic_marker_outline);
+    Drawable fill = AppCompatResources.getDrawable(context, R.drawable.ic_marker_fill);
+    Drawable overlay = AppCompatResources.getDrawable(context, R.drawable.ic_marker_overlay);
+    // TODO: Define scale in resources.
+    // TODO: Adjust size based on zoom level and selection state.
+    float scale = 1.8f;
+    int width = (int) (outline.getIntrinsicWidth() * scale);
+    int height = (int) (outline.getIntrinsicHeight() * scale);
+    Bitmap bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+    Canvas canvas = new Canvas(bitmap);
+    outline.setBounds(0, 0, width, height);
+    outline.draw(canvas);
+    fill.setColorFilter(color, PorterDuff.Mode.SRC_ATOP);
+    fill.setBounds(0, 0, width, height);
+    fill.draw(canvas);
+    overlay.setBounds(0, 0, width, height);
+    overlay.draw(canvas);
+    // TODO: Cache rendered bitmaps.
+    return BitmapDescriptorFactory.fromBitmap(bitmap);
   }
 
-  @NonNull
-  public static @DrawableRes int getResourceId(Context context, @Nullable String iconId) {
-    if (iconId == null) {
-      return R.drawable.ic_default_map_marker;
-    }
+  private static int getIconColor(Context context, String colorHexCode) {
     try {
-      String resourceName = "ic_marker_" + iconId.replace("-", "_");
-      int resourceId =
-          context.getResources().getIdentifier(resourceName, "drawable", context.getPackageName());
-      if (resourceId > 0) {
-        return resourceId;
-      }
-    } catch (Resources.NotFoundException e) {
-      Log.w(TAG, e);
+      return Color.parseColor(colorHexCode);
+    } catch (IllegalArgumentException e) {
+      return context.getResources().getColor(R.color.colorMapAccent);
```

Comment:
In addition, let's log an error as well to keep track of such events.

Reply:
Good idea! In the process I've done a little more cleanup that I'd probably never get back to otherwise: 50049aaea418d44ff6c3fb78d584947b4acb2174

Reply:
Thanks!